Aijuh/182/log with logging

Aijuh opened the pull request:
This closes #182.

I changed all codes using log channel into the codes using logging in poller.py, dbmgr.py, numger.py.
Format using 'f' is changed to '%' in logging to prevent pylint error. All logs' level is logging.info.
I think in cofig.json, channel name log should be deleted. But I leave it just in case.

Aijuh committed:
Change log channel to logging.

1. Import logging for all example code.
2. Change codes using log channel into logging. (All logging level is logging.info)

Aijuh committed:
Convert 'f format' to '% format' in logging for pylint

Aijuh committed:
blank space

BECATRUE commented on examples/datacalc.py:
```diff
@@ -137,8 +141,7 @@ def setDB(self, name: str):
         self.dbNames[name] = dbBox.currentText()
-        self.broadcast(
-            "log", 
+        logging.info(
```
Please use % formatting, too!

BECATRUE commented on examples/datacalc.py:
```diff
@@ -162,4 +165,4 @@ def calculateSum(self):
         else:
             self.viewerFrame.numberLabel.setText(f"sum: {result}")
-            self.broadcast("log", f"Sum: {result}.")
+            logger.log(f"Sum: {result}.")
```
The same as above.

BECATRUE commented on examples/numgen.py:
```diff
@@ -138,18 +142,17 @@ def receivedSlot(self, channelName: str, content: Any):
         else:
-            print(f"The message was ignored because "
-                  f"the treatment for the channel {channelName} is not implemented.")
+            print("The message was ignored because "\
```
Please remove the backslash.

BECATRUE reviewed (changes requested):
It looks good!

If possible, could you add additional logs in each app, e.g., the failure case to save number in `numgen.py`?
Of course, you can separate this into another issue!

Also, please remove the channel `log` from `config.json`.

Aijuh commented:
I updated codes. 
Deleted backslash, and changed % format.
As for error case message, all the number in logging, it has well defined range. So we cannot know error from its value. 
Otherwise I added logger messages for error cases if there was none.

Aijuh committed:
feedback

Aijuh committed:
pylint error

Aijuh committed:
blank error

BECATRUE commented:
Oh, after you finish to apply our reviews, please re-request to us by clicking the above reviewers tab!
It will make us check quickly.

BECATRUE commented on examples/datacalc.py:
```diff
@@ -137,12 +141,9 @@ def setDB(self, name: str):
-        )
+        if self.dbNames[name]:
+            logger.info("Database %s is set as %s.", name, self.dbNames[name])
+        else: logger.info("Database %s is not selected.", name)
```
Please move the body to a new line!

BECATRUE reviewed (changes requested):
I think all `print()` statements should be replaced to use logging.
Indeed, we just printed all logs before because the logger is not completed yet😅

Aijuh committed:
print to logger

A feedback reflected.
Change print to logger

Aijuh commented:
I changed `print` to `logger`. But there is some issue.

I changed 'f' format including data type dictionary `db` to the '%' format as below by importing json.
There might be bad thing by importing json or I misused it.
`logger.info("The message was ignored because "`
`"the database %s has no such key; name or path.", json.dumps(db))`

Also in examples/backend.py, I couldn't figure out what is e!r in `print(f"apps.backend.write(): {e!r}")`, so I just leave it as before.
Can I just set it as `logger.info("apps.backend.write(): %s", {e!r})`?

BECATRUE commented:
> There might be bad thing by importing json or I misused it.
`logger.info("The message was ignored because "
"the database %s has no such key; name or path.", json.dumps(db))`

I think there is no problem because `json.dumps()` returns just a string!

BECATRUE commented:
> Can I just set it as `logger.info("apps.backend.write(): %s", {e!r})?`

I'm not sure, but I think it doesn't cause a problem.
I couldn't find the detailed documentation about `!r`, but it looks another way to use `repr()`.
In other words, `e|r` is the same as `repr(e)`.

For more details, please see this [documentation](https://docs.python.org/2/tutorial/inputoutput.html#fancier-output-formatting) and this [example](https://peps.python.org/pep-0654/#unmatched-exceptions).
I think it is just a convention to print an exception.

Aijuh commented:
Thank you for the reply.
I think the type format is string but sure about it. I will leave `print` just in case.

Aijuh committed:
e!r to %r

Aijuh commented:
Thank you for good reference. I changed {e!r} to %r.

BECATRUE commented on examples/backend.py:
```diff
@@ -56,7 +60,7 @@ def read(db_path: str, table: str) -> Any:
             ).fetchone()[0]
     except sqlite3.Error as e:
-        print(f"apps.backend.read(): {e!r}")
+        logger.error("apps.backend.read(): %r", e)
```
👍

BECATRUE commented on examples/datacalc.py:
```diff
@@ -93,8 +98,8 @@ def updateDB(self, content: dict):
-                print(f"The message was ignored because "
-                        f"the database {db} has no such key; name or path.")
+                logger.info("The message was ignored because "
+                        "the database %s has no such key; name or path.", json.dumps(db))
```
Please align the starting positions of the two lines of double quotation marks.

```python
logger.info("The message was ignored because "
            "the database %s has no such key; name or path.", json.dumps(db))
```

BECATRUE commented on examples/datacalc.py:
```diff
@@ -137,12 +142,8 @@ def setDB(self, name: str):
-            else f"Database {name} is not selected."
-        )
+        logger.info("Database %s is set as %s.", (name, self.dbNames[name]) if self.dbNames[name]
+                    else "Database %s is not selected.", name)
```
Does it run well..?
In my local, it seems to match the formatting incorrectly.

kangz12345 replied:
For such cases, please use the `if-else` _statement_, not the _expression_ for readability (for all the other similar cases as well).
For example:
```python
if self.dbNames[name]:
    logger.info("Database %s is set as %s.", name, self.dbNames[name])
else:
    logger.info("Database %s is not selected.", name)
```

Aijuh replied:
I tested it and found log message printed well. But I will change the code as @kangz12345 suggested.

BECATRUE commented on examples/logger.py:
```diff
@@ -186,10 +186,10 @@ def receivedSlot(self, channelName: str, content: Any):
                 self.addLog(content)
             else:
-                print("The message for the channel log should be a string.")
+                self.logger.info("The message for the channel log should be a string.")
```
Please use `logger = logging.getLogger(__name__)` not `self.logger`!

Aijuh replied:
I remember I heard that we should not set logger in logger.py as global variable. But I will set it as global variable for now.

BECATRUE replied:
Can you explain this more?

Aijuh replied:
I don't remember exact reason. But I heard that setting root logger as global variable is something to be avoided. 
I changed my previous plan of using global variable as below. 
1. In constructor of LoggerApp, set root logger as local variable and set its level as "DEBUG" and addHandler.
2. When we need logger of logger.py (i.e. root logger), just call root logger using logging.getLogger(). (line 189, 191)

kangz12345 replied:
You can use multiple loggers. The root logger is only for _reading_ the logs.
To _write_ logs in `loggerApp`, please use `logging.getLogger(__name__)`.

BECATRUE replied:
I also agree with @kangz12345's suggestion!

BECATRUE commented on examples/numgen.py:
```diff
@@ -109,8 +114,8 @@ def updateDB(self, content: dict):
-                print(f"The message was ignored because "
-                        f"the database {db} has no such key; name or path.")
+                logger.info("The message was ignored because "
+                        "the database %s has no such key; name or path.", json.dumps(db))
```
Indentation!

BECATRUE commented on examples/numgen.py:
```diff
@@ -136,20 +141,19 @@ def receivedSlot(self, channelName: str, content: Any):
-            print(f"The message was ignored because "
-                  f"the treatment for the channel {channelName} is not implemented.")
+            logger.info("The message was ignored because "
+                  "the treatment for the channel %s is not implemented.", channelName)
```
Indentation!

BECATRUE commented on examples/numgen.py:
```diff
@@ -136,20 +141,19 @@ def receivedSlot(self, channelName: str, content: Any):
-            else "Database to store is not selected."
+        logger.info(
+            "Database to store is set as %s." if self.dbName
+            else "Database to store is not selected.", self.dbName
```
It seems to make a syntax error.

```python
logger.info(
            "Database to store is set as %s.", self.dbName if self.dbName
            else "Database to store is not selected."
)
```

Isn't this correct?

Aijuh replied:
I found 
_logger.info(
            "Database to store is set as %s." % self.dbName if self.dbName
            else "Database to store is not selected."
)_
print message as we want.

But pylint tell me to use 
_logger.info(
          f"Database to store is set as {self.dbName}." if self.dbName
            else "Database to store is not selected."
)_ 
again. So I used it.

BECATRUE replied:
Oh, then does Pylint say that my suggestion doesn't meet the Pylint condition?
I think not using the _f-formatting_ is better..

kangz12345 replied:
As I said, please use `if-else` statement instead. I don't see any benefit from using the ternary expression.
In addition, `logger.info("%s" % x)` and `logger.info("%s", x)` are different! The former is just using the `%`-formatting, which should be replaced with the f-string as pylint complained. The latter is using the `logging` API. Comment questions if you need more explanation about this.

kangz12345 replied:
In addition, previously the code used the ternary expression because it didn't use the `logging` API.

Aijuh replied:
@kangz12345 : I understand. I will use if-else instead.
@BECATRUE : When I print message as you suggested using `print` in python.idle, if `self.dbName == Sam`, it prints  _"Database to store is set as %s. Sam"_. But what we want is _"Database to store is set as Sam"_. I don't test for exact logger so I am not sure about case of logging.

BECATRUE replied:
Oh, thanks for notifying it.
But, if you adapt Jiyong's suggestion, we don't have to consider this problem more.

BECATRUE commented on examples/poller.py:
```diff
@@ -95,8 +100,8 @@ def updateDB(self, content: dict):
-                print(f"The message was ignored because "
-                        f"the database {db} has no such key; name or path.")
+                logger.info("The message was ignored because "
+                        "the database %s has no such key; name or path.", json.dumps(db))
```
Indentation!

BECATRUE commented on examples/poller.py:
```diff
@@ -122,26 +127,25 @@ def receivedSlot(self, channelName: str, content: Any):
-            else "Polled database is not selected."
+        logger.info(
+            "Polled database is set as %s." if self.dbName
+            else "Polled database is not selected.", self.dbName
```
Please check the syntax error.

BECATRUE reviewed (changes requested):
Several codes are likely to make syntax errors.
It would be a tedious work, but please check if each statement doesn't cause an error.

kangz12345 commented:
For logging the exceptions, you can consider using [`exception()`](https://docs.python.org/3/library/logging.html#logging.Logger.exception).
This will (probably) print out the exception stack information as well.

kangz12345 commented on examples/datacalc.py:
```diff
@@ -162,4 +163,4 @@ def calculateSum(self):
         else:
             self.viewerFrame.numberLabel.setText(f"sum: {result}")
-            self.broadcast("log", f"Sum: {result}.")
+            logger.log("Sum: %f.", result)
```
Why did you use `log()` instead of `info()` here?

Aijuh replied:
My mistake. I will change to log.info for integrity.

Aijuh committed:
Change format

In backend.py
    Change logger.error to logging.exception.
In datacalc.py
    Fix indentation error.(line 102)
    Divide one log block to if-else statement(line 145)
    Fix logger.log to logger.info(168)
In logger.py
    Set logger as not class variable. When logger is needed, get root logger by
    using logging.getLogger()
In numgen.py
    Fix indentation error and return to f-format as pylint wanted.
In poller.py
    Same as numger.py

Aijuh commented:
Fix codes.

BECATRUE commented on examples/datacalc.py:
```diff
@@ -123,10 +128,10 @@ def receivedSlot(self, channelName: str, content: Any):
-            print(f"The message was ignored because "
-                  f"the treatment for the channel {channelName} is not implemented.")
+            logger.info("The message was ignored because "
+                        "the treatment for the channel %s is not implemented.", channelName)
```
I think all loggings in this file should be set as `WARNING` or `ERROR`.

How about @kangz12345?

kangz12345 replied:
Oh, nice point! I wanted to point it out but I forgot.
@Aijuh please choose appropriate log levels and use `warning()`, `error()`, and `exception()` as well.

BECATRUE commented on examples/dbmgr.py:
```diff
@@ -181,7 +184,7 @@ def openCloseDatacalc(self):
         if self.openCloseDatacalcResult is not None:
             if not self.openCloseDatacalcResult.done:
-                print("DBMgrApp.openCloseDatacalc(): The previous qiwiscall must be done.")
+                logger.info("DBMgrApp.openCloseDatacalc(): The previous qiwiscall must be done.")
```
The same as the above.

BECATRUE commented on examples/numgen.py:
```diff
@@ -109,8 +114,8 @@ def updateDB(self, content: dict):
-                print(f"The message was ignored because "
-                        f"the database {db} has no such key; name or path.")
+                logger.info("The message was ignored because "
+                            "the database %s has no such key; name or path.", json.dumps(db))
```
The same as the above.

BECATRUE replied:
Indentation!

BECATRUE commented on examples/numgen.py:
```diff
@@ -161,12 +165,13 @@ def generateNumber(self):
+            logger.info("Generated number saved.")
         else:
             self.viewerFrame.statusLabel.setText("failed to save number")
+            logger.info("Failed to save generated number")
```
The same as the above.

BECATRUE commented on examples/numgen.py:
```diff
@@ -136,18 +141,17 @@ def receivedSlot(self, channelName: str, content: Any):
-            print(f"The message was ignored because "
-                  f"the treatment for the channel {channelName} is not implemented.")
+            logger.info("The message was ignored because "
+                        "the treatment for the channel %s is not implemented.", channelName)
```
The same as the above.

BECATRUE replied:
Indentation!

BECATRUE commented on examples/numgen.py:
```diff
@@ -136,18 +141,17 @@ def receivedSlot(self, channelName: str, content: Any):
                 self.updateDB(content)
             else:
-                print("The message for the channel db should be a dictionary.")
+                logger.info("The message for the channel db should be a dictionary.")
```
The same as the above.

BECATRUE commented on examples/poller.py:
```diff
@@ -95,8 +100,8 @@ def updateDB(self, content: dict):
             if any(key not in db for key in ("name", "path")):
-                print(f"The message was ignored because "
-                        f"the database {db} has no such key; name or path.")
+                logger.info("The message was ignored because "
```
The level of all loggings of this file should be determined.

BECATRUE reviewed (changes requested):
I think we should talk about the level of all loggings, because in the most cases, we left a log for notifying a warning, not just for an information.

Also, please check the above two reviews I left before.

Aijuh commented:
I agree with that. I will wait @kangz12345's comment.

Aijuh committed:
log level select

Aijuh commented:
I refer to [logging](https://docs.python.org/3/howto/logging.html) 's log level.
I understand error as problem in a particular function, and warning as possible problem.
So  In the code below I choose warning because two case has same function for now.
```python
if self.dbNames[name]:
    logger.info("Database %s is set as %s.", name, self.dbNames[name])
else:
    logger.warning("Database %s is not selected.", name)

```

In the code below I use error because it has different function at that time.
```python
for db in content.get("db", ()):
    if any(key not in db for key in ("name", "path")):
        logger.error("The message was ignored because "
                     "the database %s has no such key; name or path.", json.dumps(db))
        continue
```

Aijuh committed:
pylint..

kangz12345 commented:
> I refer to [logging](https://docs.python.org/3/howto/logging.html) 's log level. I understand error as problem in a particular function, and warning as possible problem. So In the code below I choose warning because two case has same function for now.
> 
> ```python
> if self.dbNames[name]:
>     logger.info("Database %s is set as %s.", name, self.dbNames[name])
> else:
>     logger.warning("Database %s is not selected.", name)
> ```

Actually in that case, I think `info()` is okay because it is a valid expected input (you can _unselect_ the database).
Of course, after all, databases should be determined so you can label it as a warning.
Conclusion: you can decide which to use. I just wanted to tell you that it is not an unexpected situation.

Aijuh committed:
warning to info

Aijuh commented:
Changed warning to info.

Aijuh committed:
fix logger in logger.py

BECATRUE commented on examples/datacalc.py:
```diff
@@ -93,8 +98,8 @@ def updateDB(self, content: dict):
-                print(f"The message was ignored because "
-                        f"the database {db} has no such key; name or path.")
+                logger.error("The message was ignored because "
+                            "the database %s has no such key; name or path.", json.dumps(db))
```
Indentation!

Aijuh replied:
sorry...

BECATRUE commented on examples/datacalc.py:
```diff
@@ -123,10 +128,10 @@ def receivedSlot(self, channelName: str, content: Any):
-            print(f"The message was ignored because "
-                  f"the treatment for the channel {channelName} is not implemented.")
+            logger.error("The message was ignored because "
+                        "the treatment for the channel %s is not implemented.", channelName)
```
Indentation!

BECATRUE commented on examples/logger.py:
```diff
@@ -186,10 +186,10 @@ def receivedSlot(self, channelName: str, content: Any):
                 self.addLog(content)
             else:
-                print("The message for the channel log should be a string.")
+                (logging.getLogger(__name__)).error("The message for the channel log should be a string.")
```
Please define `logging.getLogger(__name__)` as a global variable like other files.

BECATRUE commented on examples/logger.py:
```diff
@@ -186,10 +186,10 @@ def receivedSlot(self, channelName: str, content: Any):
-            print(f"The message was ignored because "
-                  f"the treatment for the channel {channelName} is not implemented.")
+            (logging.getLogger(__name__)).error("The message was ignored because "
+                             "the treatment for the channel %s is not implemented.", channelName)
```
Indentation!

BECATRUE commented on examples/numgen.py:
```diff
@@ -136,21 +141,19 @@ def receivedSlot(self, channelName: str, content: Any):
-        )
+        if self.dbName:
+            logger.info("Database to store is set as %s", self.dbName)
+        else: logger.info("Database to store is not selected.")
```
Please write the body statement to a new line.

BECATRUE commented on examples/poller.py:
```diff
@@ -122,27 +127,25 @@ def receivedSlot(self, channelName: str, content: Any):
-        )
+        if self.dbName:
+            logger.info("Database to store is set as %s.", self.dbName)
+        else: logger.warning("Database to store is not selected.")
```
New line!

Also, in `numgen.py`, you handled this case with `logger.info()`.
Please unify both statements of two files.

BECATRUE reviewed (changes requested):
Please check indentations!

Aijuh committed:
Indentation&global logger in logger.py

Aijuh commented:
I will re-read changed code for later. Sorry for repeated error.

Aijuh committed:
fix-error

Aijuh committed:
Update poller.py

BECATRUE commented:
> I will re-read changed code for later. Sorry for repeated error.

It's okay😃 I will review the updated parts.

BECATRUE commented on examples/dbmgr.py:
```diff
@@ -181,7 +184,7 @@ def openCloseDatacalc(self):
         if self.openCloseDatacalcResult is not None:
             if not self.openCloseDatacalcResult.done:
-                print("DBMgrApp.openCloseDatacalc(): The previous qiwiscall must be done.")
+                logger.info("DBMgrApp.openCloseDatacalc(): The previous qiwiscall must be done.")
```
How about using `logging.warning()`?
If you think `logging.info()` is more appropriate, never mind!

kangz12345 replied:
I think we can remove the header "DBMgrApp.openCloseDatacalc():" (for the other similar cases as well). Do you agree @BECATRUE?

kangz12345 replied:
@BECATRUE @Aijuh

BECATRUE commented on examples/poller.py:
```diff
@@ -95,8 +100,8 @@ def updateDB(self, content: dict):
-                print(f"The message was ignored because "
-                        f"the database {db} has no such key; name or path.")
+                logger.error("The message was ignored because "
+                            "the database %s has no such key; name or path.", json.dumps(db))
```
Please check the indentation.

BECATRUE commented on examples/poller.py:
```diff
@@ -122,27 +127,26 @@ def receivedSlot(self, channelName: str, content: Any):
-            print(f"The message was ignored because "
-                  f"the treatment for the channel {channelName} is not implemented.")
+            logger.error("The message was ignored because "
+                        "the treatment for the channel %s is not implemented.", channelName)
```
Here, too!

BECATRUE commented on examples/logger.py:
```diff
@@ -186,10 +189,10 @@ def receivedSlot(self, channelName: str, content: Any):
         else:
-            print(f"The message was ignored because "
-                  f"the treatment for the channel {channelName} is not implemented.")
+            logger.error("The message was ignored because "
```
I just realized it now, and I think you can remove `receivedSlot()` becuase we don't use the `log` channel more.

Aijuh replied:
I knew that but just leave it. I will delete receivedSlot.

BECATRUE replied:
I'm sorry I should have told you earlier😅

Aijuh replied:
Anytime.

BECATRUE reviewed (changes requested):
It looks almost done😃

I left only a few reviews.

Aijuh committed:
fix_error&delete received Slot in loger.py

Aijuh commented:
Code fixed.

Aijuh committed:
broadcast to logging

kangz12345 commented on examples/backend.py:
```diff
@@ -56,7 +60,7 @@ def read(db_path: str, table: str) -> Any:
             ).fetchone()[0]
     except sqlite3.Error as e:
-        print(f"apps.backend.read(): {e!r}")
+        logging.exception("apps.backend.read(): %r", e)
```
Use `logger`, not `logging`.
Moreover, since `exception()` prints out the stack trace, you don't have to print the exception.
Instead, please describe what has failed.
For example: 
```python
logger.exception("Failed to read table %s from database %s.", table, db_path)
```

kangz12345 commented on examples/dbmgr.py:
```diff
@@ -121,10 +125,9 @@ def sendDB(self, isAdded: bool, name: str):
-            "log",
-            f"Database {name} is added." if isAdded
-            else f"Database {name} is removed."
+        logger.info(
+            "Database %s is added." if isAdded
+            else "Database %s is removed.", name
         )
```
Another way to write it:
```python
logger.info("Database %s is %s.", name, "added" if isAdded else "removed")
```
You can use your previous code! Just follow your preference.

Aijuh replied:
I think this suggestion is easy to read. I will change in this way.

kangz12345 commented on examples/logger.py:
```diff
@@ -135,9 +138,9 @@ def __init__(self, name: str, parent: Optional[QObject] = None):
-        logger.addHandler(self.handler)
+        rootLogger = logging.getLogger()
+        rootLogger.setLevel(logging.DEBUG)
+        rootLogger.addHandler(self.handler)
```
👍

kangz12345 commented on examples/logger.py:
```diff
@@ -174,27 +177,10 @@ def addLog(self, content: str):
     def checkToClear(self):
         """Shows a confirmation frame for log clearing."""
-        self.broadcast("log", "Clicked to clear logs")
+        logger.info("Clicked to clear logs")
```
Please append a period at the end.

kangz12345 commented on examples/numgen.py:
```diff
@@ -161,12 +165,13 @@ def generateNumber(self):
+            logger.info("Generated number saved.")
         else:
             self.viewerFrame.statusLabel.setText("failed to save number")
+            logger.error("Failed to save generated number")
```
Please append a period at the end.

kangz12345 commented on examples/poller.py:
```diff
@@ -151,8 +155,10 @@ def poll(self):
         if write(os.path.join(dbPath, self.dbName), self.table, num):
-            self.broadcast("log", "Polled number saved.")
+            logger.info("Polled number saved.")
+        else:
+            logger.error("Failed to save polled number.")
```
👍

Aijuh committed:
Fixed format error

Aijuh committed:
Update string format.

Aijuh committed:
Update logger.py

BECATRUE reviewed (changes requested):
Please apply the remaining parts that we discussed in snu-quiqcl/iquip#74.

kangz12345 commented:
> Please apply the remaining parts that we discussed in [snu-quiqcl/iquip#74](https://github.com/snu-quiqcl/iquip/issues/74).

Please describe the request in more detail.

Aijuh commented:
I understand the request. I will update qiwis with respect to iquip.

Aijuh commented:
I will first update until currently merged `develop` branch of iquip. Give me comments if you want further update.

Aijuh committed:
Update with respect to iquip's logger.py

BECATRUE reviewed (dismissed):
LGTM!

Please request @kangz12345 for reviews.

kangz12345 commented on examples/backend.py:
```diff
@@ -99,8 +99,8 @@ def write(db_path: str, table: str, value: Any) -> bool:
-    except sqlite3.Error as e:
-        logging.exception("apps.backend.write(): %r", e)
+    except sqlite3.Error:
+        logger.exception("Failed to write database %s from table %s.", db_path, table)
```
Since the table is in the database, this expression seems awkward.
How about `"Failed to write a value into table %s of database %s"`?

Aijuh replied:
I don't know details about database. I will fix it in this way.

Aijuh committed:
Updated feedback from @kangz12345

Aijuh commented:
Fixed the code.

kangz12345 commented on examples/backend.py:
```diff
@@ -100,7 +100,7 @@ def write(db_path: str, table: str, value: Any) -> bool:
             )
     except sqlite3.Error:
-        logger.exception("Failed to write database %s from table %s.", db_path, table)
+        logger.exception("Failed to write a value into table %s of database %s", db_path, table)
```
😄 The order of the following arguments should be switched as well!

Aijuh committed:
Fixed variable order mistake.

Aijuh commented:
I fixed it. I should have read it.

kangz12345 reviewed (approved):
LGTM

BECATRUE reviewed (approved):
LGTM
